sdcm/cluster_k8s/gke.py

```diff
@@ -0,0 +1,185 @@
+        self.gcloud(f"container --project {self.gce_project} clusters create {self.name} --username admin "
+                    f"--zone {self.gce_zone} --cluster-version {self.gke_cluster_version} "
+                    f"--machine-type {self.gce_instance_type} --num-nodes {num_nodes} "
+                    f"--disk-type {self.gce_image_type} --disk-size {self.gce_image_size} "
+                    f"--local-ssd-count {self.gce_n_local_ssd} --node-taints role=scylla-clusters:NoSchedule "
+                    f"--image-type UBUNTU --enable-stackdriver-kubernetes --no-enable-autoupgrade "
+                    f"--no-enable-autorepair --metadata {tags} --network {self.gce_network}")
+        self.gcloud(f"container --project {self.gce_project} node-pools create operator-pool "
+                    f"--cluster {self.name} --zone {self.gce_zone} --machine-type n1-standard-4 --num-nodes 1 "
+                    f"--disk-type pd-ssd --disk-size 20 --image-type UBUNTU --no-enable-autoupgrade "
+                    f"--no-enable-autorepair")
+
+        LOGGER.info("Get credentials for GKE cluster `%s'", self.name)
+        self.gcloud(f"container clusters get-credentials {self.name} --zone {self.gce_zone}")
+        self.patch_kube_config()
+
+        LOGGER.info("Setup RBAC for GKE cluster `%s'", self.name)
+        self.kubectl(
+            f"create clusterrolebinding cluster-admin-binding --clusterrole cluster-admin --user {self.gce_user}")
+
+        LOGGER.info("Install RAID DaemonSet to GKE cluster `%s'", self.name)
+        self.apply_file(RAID_DAEMONSET, envsubst=False)
+
+        LOGGER.info("Install CPU policy DaemonSet to GKE cluster `%s'", self.name)
+        self.apply_file(CPU_POLICY_DAEMONSET, envsubst=False)
+
+        LOGGER.info("Install local volume provisioner to GKE cluster `%s'", self.name)
+        self.helm(f"install local-provisioner provisioner")
+
+    def patch_kube_config(self) -> None:
+        kube_config_path = os.path.expanduser("~/.kube/config")
+
+        LOGGER.debug("Patch %s to use dockerized gcloud for auth against GKE cluster `%s'", kube_config_path, self.name)
+        with open(kube_config_path) as kube_config:
+            data = yaml.safe_load(kube_config)
+
+        user_name = f"gke_{self.gce_project}_{self.gce_zone}_{self.name}"
+        for user in data["users"]:
+            if user["name"] == user_name:
+                config = user["user"]["auth-provider"]["config"]
+                break
+        else:
+            raise RuntimeError(f"Unable to find configuration for `{user_name}' in ~/.kube/config")
+
+        config["cmd-args"] = f"exec {self.gcloud_container_id} gcloud config config-helper --format=json"
+        config["cmd-path"] = "/usr/bin/docker"
+
+        with open(kube_config_path, "w") as kube_config:
+            yaml.safe_dump(data, kube_config)
+
+    @cluster.wait_for_init_wrap
+    def wait_for_init(self):
+        pass
+
+    def destroy(self):
+        pass
```

Comment:
How do we cleanup resources used by this guy ?

Reply:
We don't handle GKE cluster nodes in SCT for now. So, there are no resources that we need to care about.